src/main/java/com/wanderersoftherift/wotr/core/OnHitEvent.java

```diff
@@ -0,0 +1,18 @@
+package com.wanderersoftherift.wotr.core;
+
+import com.wanderersoftherift.wotr.WanderersOfTheRift;
+import net.minecraft.world.entity.LivingEntity;
+import net.neoforged.bus.api.SubscribeEvent;
+import net.neoforged.fml.common.EventBusSubscriber;
+import net.neoforged.neoforge.event.entity.living.LivingIncomingDamageEvent;
+
+@EventBusSubscriber(modid = WanderersOfTheRift.MODID, bus = EventBusSubscriber.Bus.GAME)
+public class OnHitEvent {
+
+    @SubscribeEvent
+    public static void onHit(LivingIncomingDamageEvent event) {
+        float dmg = OnHitEffect.critical(event.getAmount(), (LivingEntity) event.getSource().getDirectEntity(),
```

Comment:
A technique I like to use is 
```java
if (event.getSource().getDirectEntity() instanceof LivingEntity livingAttacker) {
  // proceed using livingAttacker.   
}
```
This combines checking the directEntity is not null and is the right type with casting it.